client/my-sites/themes/theme-showcase.jsx

```diff
@@ -48,15 +49,41 @@ const ThemeShowcase = React.createClass( {
 		this.setState( { showPreview: ! this.state.showPreview, previewingTheme: theme } );
 	},
 
-	onPreviewButtonClick( theme ) {
-		const { defaultOption } = this.props;
+	onPrimaryPreviewButtonClick( theme ) {
+		const option = this.getPrimaryOption();
 		this.setState( { showPreview: false }, () => {
-			defaultOption.action && defaultOption.action( theme );
+			option.action && option.action( theme );
 		} );
 	},
 
+	onSecondaryPreviewButtonClick( theme ) {
+		const { secondaryOption } = this.props;
+		this.setState( { showPreview: false }, () => {
+			secondaryOption && secondaryOption.action ? secondaryOption.action( theme ) : null;
+		} );
+	},
+
+	getPrimaryOption() {
```

Comment:
Does this have some of the same logic as the theme info sheet here? https://github.com/Automattic/wp-calypso/blob/2d76fa954a0e1dc13f3302345280e1e14fe3f607/client/my-sites/theme/main.jsx#L397

If so, it would be better if we could share it somehow, maybe by adding a function to the theme helpers https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/helpers.js